scene/gui/text_edit.cpp

```diff
@@ -7434,8 +7453,8 @@ void TextEdit::_update_scrollbars() {
 		total_width += minimap_width;
 	}
 
-	content_height_cache = MAX(total_rows, 1) * get_line_height();
-	if (fit_content_height) {
+	content_size_cache = Vector2i(text.get_max_width() + 10, MAX(total_rows, 1) * get_line_height());
```

Comment:
```suggestion
	content_size_cache = Vector2i(total_width + 10, MAX(total_rows, 1) * get_line_height());
```

This should use the total_width calculated above instead of the text width, otherwise the hscrollbar is shown when the minimap, any gutters, or placeholders are drawn.